contracts/AMMWrapper.sol

```diff
@@ -219,29 +222,19 @@ contract AMMWrapper is IAMMWrapper, ReentrancyGuard, BaseLibEIP712, SignatureVal
 
         _prepare(order, internalTxData);
 
-        // minAmount = makerAssetAmount * (10000 - subsidyFactor) / 10000
-        uint256 _minAmount = order.makerAssetAmount.mul((LibConstant.BPS_MAX.sub(txMetaData.subsidyFactor))).div(LibConstant.BPS_MAX);
-        (txMetaData.source, txMetaData.receivedAmount) = _swap(order, internalTxData, _minAmount);
+        {
+            // Set min amount for swap = _order.makerAssetAmount * (1 + feeFactor)
+            uint256 swapMinOutAmount = order.makerAssetAmount.mul(LibConstant.BPS_MAX.add(txMetaData.feeFactor).div(LibConstant.BPS_MAX));
+            (txMetaData.source, txMetaData.receivedAmount) = _swap(order, internalTxData, swapMinOutAmount);
 
-        // Settle
-        txMetaData.settleAmount = _settle(order, txMetaData, internalTxData);
-
-        emit Swapped(
-            txMetaData.source,
-            txMetaData.transactionHash,
-            order.userAddr,
-            order.takerAssetAddr,
-            order.takerAssetAmount,
-            order.makerAddr,
-            order.makerAssetAddr,
-            order.makerAssetAmount,
-            order.receiverAddr,
-            txMetaData.settleAmount,
-            txMetaData.receivedAmount,
-            txMetaData.feeFactor,
-            txMetaData.subsidyFactor
-        );
+            // Settle
+            // Calculate fee using actually received from swap
+            uint256 actualFee = txMetaData.receivedAmount.mul(txMetaData.feeFactor).div(LibConstant.BPS_MAX);
```

Comment:
The fee will be different from the one computed above (in line 227). If this is intended (i.e., we can collect more fee if received more), we should also update the spec.
Otherwise we should probably store the fee in a local variable.

Reply:
This is intended and aligned with the spec. The reason of this is to prevent user bringing too small `makerAssetAmount` although it's not likely.

Reply:
What I'm mean is that the `actualFee` computed in `_swap` and `_settle` would be different while in the spec it's like this:
```solidity
uint256 fee = _order.makerAssetAmount.mul(feeFactor).div(BPS_MAX);
// AMM should output with minAmount = makerAssetAmount + fee 
uint256 minAmount = _order.makerAssetAmount.add(fee);
uint256 receivedAmount = _swap(_order, minAmount);

// Transfer (receivedAmount - fee) to user
uint256 settleAmount = receivedAmount.sub(fee);
token.safeTransfer(_order.receiverAddr, settleAmount);
```

so we should probably align them.

Reply:
My bad. I think the spec should be updated if we all accept the logic in this PR.

Reply:
I support this one too. Let's update the spec.

Reply:
Gogogo

Reply:
No wonder I'm a little bit confused. If so, it might be involved with changing FE logic. For the app, it needs to handle two different logics from old and new contract?

Reply:
I'll confirm with FE team.